eng/common/cross/toolchain.cmake

```diff
@@ -43,10 +43,30 @@ if(TARGET_ARCH_NAME STREQUAL "armel")
   endif()
 endif()
 
-set(CMAKE_SYSROOT "${CROSS_ROOTFS}")
-set(CMAKE_C_COMPILER_EXTERNAL_TOOLCHAIN "${CROSS_ROOTFS}/usr")
-set(CMAKE_CXX_COMPILER_EXTERNAL_TOOLCHAIN "${CROSS_ROOTFS}/usr")
-set(CMAKE_ASM_COMPILER_EXTERNAL_TOOLCHAIN "${CROSS_ROOTFS}/usr")
+if("$ENV{__DistroRid}" MATCHES "android.*")
+    if(TARGET_ARCH_NAME STREQUAL "arm")
+        set(ANDROID_ABI armeabi-v7a)
+    elseif(TARGET_ARCH_NAME STREQUAL "arm64")
+        set(ANDROID_ABI arm64-v8a)
+    endif()
+
+    # extract platform number required by the NDK's toolchain
+    string(REGEX REPLACE ".*\\.([0-9]+)-.*" "\\1" ANDROID_PLATFORM "$ENV{__DistroRid}")
```

Comment:
@janvorli, `ANDROID_PLATFORM` number is required by the NDK which expects caller to specify it before calling `android.toolchain.cmake`. We have two options, `echo` the value `--cmakeargs -DANDROID_PLATFORM=$__ApiLevel` it in the last line of rootfs script: https://github.com/dotnet/runtime/blob/05d7a5b9d2c014dd9c16b1fb29be29986c0e58ca/eng/common/cross/build-android-rootfs.sh#L117-L120

or use this parse-number-from-RID approach, which I have used since it is self-contained and we are using RID in other parts of the build that we have written in `$__ToolchainDir/sysroot/android_platform`.